lib/express/saml.js

```diff
@@ -95,18 +95,27 @@ function config(
             xml,
           )
           console.warn(`Received SAML Artifact ${samlArtifact}`)
-          // Take the template and plug in the typical SingPass/CorpPass response
-          // Sign and encrypt the assertion
-          const samlArtifactBuffer = Buffer.from(samlArtifact, 'base64')
-          let index = samlArtifactBuffer.readInt8(samlArtifactBuffer.length - 1)
-          // use env NRIC when SHOW_LOGIN_PAGE is false
-          if (index === -1) {
-            index =
-              idp === 'singPass'
-                ? assertions.saml.singPass.indexOf(assertions.singPassNric)
-                : assertions.saml.corpPass.findIndex(
-                    (c) => c.nric === assertions.corpPassNric,
-                  )
+          let nric = samlArtifact
+
+          // if not custom NRIC,
+          if (samlArtifact.length > 9) {
```

Comment:
I'm not comfortable with this - Could you document the condition? It isn't immediately obvious that if `samlArtifact.length === 9` then it must be an NRIC of some form

Reply:
Thanks, I should have been clearer in the comments